client/login/wp-login/login-links.jsx

```diff
@@ -67,26 +69,31 @@ export class LoginLinks extends React.Component {
 	};
 
 	renderBackLink() {
-		const { backTo, translate } = this.props;
-
-		// A backTo prop may be supplied, allowing the back button href to be controlled.
-		if ( backTo ) {
-			const { hostname, protocol } = urlModule.parse( backTo );
+		// If we seem to be in a Jetpack connection flow, provide some special handling
+		// so users can go back to their site rather than WordPress.com
+		const redirectTo = get( this.props, [ 'query', 'redirect_to' ] );
+		if ( redirectTo ) {
+			const { pathname, query: redirectToQuery } = url.parse( redirectTo, true );
+			if ( pathname === '/jetpack/connect/authorize' && redirectToQuery.client_id ) {
+				const returnToSiteUrl = addQueryArgs(
+					{ client_id: redirectToQuery.client_id },
+					'https://jetpack.wordpress.com/jetpack.returntosite/1/'
+				);
 
-			// Ensure we've got a relative URL (null) or an http[s] protocol
-			// We don't want to allow `javascript:…`
-			if ( null === protocol || 'http:' === protocol || 'https:' === protocol ) {
+				const { hostname } = url.parse( redirectToQuery.site_url );
```

Comment:
This will not contain the subdirectory for any subdirectory installs - for example `https://yourgroovydomain.com/subdir` will be displayed just `yourgroovydomain.com` - is that intentional?

Reply:
Not really intentional, but this uses `hostname` so it's probably irrelevant (no path). If `home_url` is more accurate I'm happy to swap it 👍

Reply:
Maybe it's a good idea to stick with the hostname - there isn't to much space here anyway.